### PR TITLE
Documentation sidebar menu

### PR DESCRIPTION
**Pull request by ericyliu**
Screenshot 2023-12-14 at 7 39 18 PM

menu

### CODE REVIEW

File: website2/src/app/docs/[slug]/page.tsx [hunk 1]
```diff
@@ -1,21 +1,49 @@
...
+    });
+    const link = filenames[i].split(".mdx")[0];
+    metadata.push({
+      index: meta.frontmatter.index,
+      name: (meta.frontmatter.name as string) || link.split("_").join(" "),
+      link,
+    });
+  }
+  return metadata.sort((a, b) => a.index - b.index);
+};
```
Comment (alexdavid) [inline website2/src/app/docs/[slug]/page.tsx:45]: But what do you think about DRYing up the fetching of the docs and slug? This way we know the static pages and links are always in sync?

Up to you, TIOLI!

```suggestion
const getDocs = async (): Promise<Array<{ fileName: string, slug: string }>> => {
  const fileNames = await readdir("documentation");
  return fileNames.map((fileName) => ({
    fileName,
    slug: fileName.split(".mdx")[0],
  }));
}

const fetchLinks = async (): Promise<Link[]> => {
  const docs = await getDocs();
  const links = await Promise.all(docs.map(async ({ fileName, slug }) => {
    const file = await readFile(`documentation/${fileName}`, "utf-8");
    const meta = await serialize<
      null,
      { index: number; name: string; link: string }
    >(file, {
      parseFrontmatter: true,
    });
    return {
      index: meta.frontmatter.index,
      name: meta.frontmatter.name || slug.replaceAll("_", " "),
      link: slug,
    };
  }));
  return links.sort((a, b) => a.index - b.index);
};

export const generateStaticParams = async () => {
  return await getDocs();
};

```

**Review by alexdavid — APPROVED** [review]
This looks good modulo comments. There is a type error on CI currently. I get a matching runtime error when visiting `/docs` so I think maybe `src/app/docs/page.tsx` needs to be updated for the new props on `DocumentationHeader`.

**Comment by alexdavid** [comment]
Ahh I see #493 fixes the error on CI

File: website2/src/app/docs/[slug]/container.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { useState } from "react";
+
+type Props = {
+  source: string;
+  links: Link[];
+  active: string;
+};
+
+export const Container = ({ source, links, active }: Props) => {
+  const [isHeaderOpen, setIsHeaderOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (ericyliu) [reply website2/src/app/docs/[slug]/container.tsx:16]: nothing, i can get rid of this cocmponent, it was being used earlier, but not anymore

File: website2/public/menu.svg
[the remark this answers is not in the document]
Comment (ericyliu) [reply website2/public/menu.svg:1]: it only needs to go in public if we want to use it as an image. If we want to inline it as an svg, I can turn it into a react component. Good point about the caches, i'll make them importable from the component directory

File: website2/src/app/docs/[slug]/page.tsx (hunk 1 above)
Comment (ericyliu) [reply website2/src/app/docs/[slug]/page.tsx:45]: great idea, I'll make this change

**Pull request merged**